dataproc/python-api-walkthrough.md

```diff
@@ -0,0 +1,158 @@
+# Use the Python Client Library to call Cloud Dataproc APIs
+
+Estimated completion time: <walkthrough-tutorial-duration duration="5"></walkthrough-tutorial-duration>
+
+## Overview
+
+This [Cloud Shell](https://cloud.google.com/shell/docs/) walkthrough leads you
+through the steps to use the
+[Google APIs Client Library for Python](http://code.google.com/p/google-api-python-client/ )
+to programmatically interact with [Cloud Dataproc](https://cloud.google.com/dataproc/docs/).
+
+As you follow this walkthrough, you run Python code that calls
+[Cloud Dataproc REST API](https://cloud.google.com//dataproc/docs/reference/rest/)
+methods to:
+
+* create a Cloud Dataproc cluster
+* submit a small PySpark word sort job to run on the cluster
+* get job status
+* tear down the cluster after job completion
+
+## Using the walkthrough
+
+The `submit_job_to_cluster.py file` used in this walkthrough is opened in the
+Cloud Shell editor when you launch the walkthrough. You can view
+the code as your follow the walkthrough steps.
+
+**For more information**: See [Cloud Dataproc&rarr;Use the Python Client Library](https://cloud.google.com/dataproc/docs/tutorials/python-library-example) for
+an explanation of how the code works.
+
+**To reload this walkthrough:** Run the following command from the
+`~/python-docs-samples/dataproc` directory in Cloud Shell:
+
+    cloudshell launch-tutorial python-api-walkthrough.md
+
+**To copy and run commands**: Click the "Paste in Cloud Shell" button
+  (<walkthrough-cloud-shell-icon></walkthrough-cloud-shell-icon>)
+  on the side of a code box, then press `Enter` to run the command.
+
+## Prerequisites (1)
+
+1. Create or select a Google Cloud Platform project to use for this tutorial.
+    * <walkthrough-project-billing-setup permissions=""></walkthrough-project-billing-setup>
+
+1. Enable the Cloud Dataproc, Compute Engine, and Cloud Storage APIs in your project.
+    * <walkthrough-enable-apis apis="dataproc,compute_component,storage-component.googleapis.com"></walkthrough-enable-apis>
+
+## Prerequisites (2)
+
+1. This walkthrough uploads a PySpark file (`pyspark_sort.py`) to a
+   [Cloud Storage bucket](https://cloud.google.com/storage/docs/key-terms#buckets) in
+   your project.
+   * You can use the [Cloud Storage browser page](https://console.cloud.google.com/storage/browser)
+   in Google Cloud Platform Console to view existing buckets in your project.
+   * To create a new bucket, run the following command. Your bucket name must be unique.
```

Comment:
Could we add a big "OR" to this? I mindlessly pasted the command to create a bucket when I already have one that I like to use for tutorials.

Reply:
Done.

Reply:
Done.